pitest/src/main/java/org/pitest/mutationtest/engine/gregor/mutators/experimental/BigDecimalMutator.java

```diff
@@ -0,0 +1,167 @@
+package org.pitest.mutationtest.engine.gregor.mutators.experimental;
+
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.pitest.mutationtest.engine.MutationIdentifier;
+import org.pitest.mutationtest.engine.gregor.MethodInfo;
+import org.pitest.mutationtest.engine.gregor.MethodMutatorFactory;
+import org.pitest.mutationtest.engine.gregor.MutationContext;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public enum BigDecimalMutator implements MethodMutatorFactory {
+  INSTANCE;
+
+  @Override
+  public MethodVisitor create(MutationContext context, MethodInfo info, MethodVisitor visitor) {
+    return new BigDecimalMathMutator(this, context, visitor);
+  }
+
+  @Override
+  public String getGloballyUniqueId() {
+    return this.getClass().getName();
+  }
+
+  @Override
+  public String toString() {
+    return "EXPERIMENTAL_BIG_DECIMAL";
+  }
+
+  @Override
+  public String getName() {
+    return toString();
+  }
+
+  private static final class BigDecimalMathMutator extends MethodVisitor {
+
+    private static final Map<String, Replacement> REPLACEMENTS;
+
+    static {
+      Map<String, Replacement> map = new HashMap<>();
+
+      String unary = "(Ljava/math/BigDecimal;)Ljava/math/BigDecimal;";
+      put(map, new Replacement("add", "subtract", unary));
+      put(map, new Replacement("subtract", "add", unary));
+      put(map, new Replacement("multiply", "divide", unary));
+      put(map, new Replacement("divide", "multiply", unary));
+      put(map, new Replacement("remainder", "multiply", unary));
+
+      put(map, new Replacement("max", "min", unary));
+      put(map, new Replacement("min", "max", unary));
+
+      String noParams = "()Ljava/math/BigDecimal;";
+      put(map, new Replacement("negate", "plus", noParams));
+      put(map, new Replacement("plus", "negate", noParams));
+      put(map, new Replacement("abs", "negate", noParams));
+
+      REPLACEMENTS = Collections.unmodifiableMap(map);
+    }
+
+    private static void put(Map<String, Replacement> map, Replacement replacement) {
+      map.put(replacement.sourceName, replacement);
+    }
+
+    private final MethodMutatorFactory factory;
+    private final MutationContext context;
+    private final String expectedOwner = "java/math/BigDecimal";
+
+    private BigDecimalMathMutator(MethodMutatorFactory factory, MutationContext context,
+                                  MethodVisitor visitor) {
+      super(Opcodes.ASM6, visitor);
+
+      this.factory = factory;
+      this.context = context;
+    }
+
+    @Override
+    public void visitMethodInsn(int opcode, String owner, String name, String descriptor,
+        boolean isInterface) {
+      if (!owner.equals(expectedOwner) || opcode != Opcodes.INVOKEVIRTUAL) {
+        this.mv.visitMethodInsn(opcode, owner, name, descriptor, isInterface);
+        return;
```

Comment:
Returning without calling visitMethodInsn will corrupt the chain for mutators that come after this. This looks to also be a bug in the BigIntegeger mutator from which this has been copied.

The issue would be revealed if this mutator were enabled in a test with others that are called after it.